app/lib/import_export.rb

```diff
@@ -10,501 +11,199 @@ module ImportExport # rubocop:todo Metrics/ModuleLength
+
+    # NOTE: in_batches appears to NOT sort within batches, so explicit ordering on ID is also done deeper down.
+    # The reorder('') here allows this ordering done later on to work properly.
+    patients.reorder('').in_batches(of: inner_batch_size).each_with_index do |batch_group, inner_batch_index|
+      # 1) Create excel file(s) on the first inner batch iteration of each outer batch. If `outer_batch_size` is 10,000 and `inner_batch_size` is 100,
+      #    then this will create a new excel file(s) every 100 batches, therefore handling the outer batching.
+      if (inner_batch_index % inner_batch_iterations_per_outer_batch).zero?
+        # One file for all data types, each data type in a different tab
+        workbooks = {} if separate_files
+        workbook = FastExcel.open(constant_memory: true) unless separate_files
+        sheets = {}
+        last_row_nums = {}
+        data_types.each do |data_type|
+          workbook = FastExcel.open(constant_memory: true) if separate_files
+          worksheet = workbook.add_worksheet(config.dig(:data, data_type, :tab) || CUSTOM_EXPORT_OPTIONS.dig(data_type, :label))
+          worksheet.auto_width = true
+          worksheet.append_row(field_data.dig(data_type, :headers))
+          last_row_nums[data_type] = 0
+          sheets[data_type] = worksheet
+          workbooks[data_type] = workbook if separate_files
+        end
+      end
+
+      # 2) Get export data in batches to decrease size of export data hash maintained in memory
+      exported_data = get_export_data(batch_group.order(:id), config[:data], symptom_names)
+      data_types.each do |data_type|
+        last_row_nums[data_type] = write_xlsx_rows(exported_data, data_type, sheets[data_type], field_data[data_type][:checked], last_row_nums[data_type])
+      end
+
+      # 3) Check if the next inner_batch_index is the start of a new outer batch or if this current batch is the final batch
+      #    If so, save files to db as this is the end of the current outer batch
+      next unless ((inner_batch_index + 1) % inner_batch_iterations_per_outer_batch).zero? || inner_batch_index == total_inner_batches - 1
+
+      outer_batch_index = inner_batch_index * inner_batch_size / outer_batch_size
+      if separate_files
+        data_types.each do |data_type|
+          file = { filename: build_export_filename(config, data_type, outer_batch_index, false), content: Base64.encode64(workbooks[data_type].read_string) }
+          files << save_file(config, file)
+        end
+      else
+        file = { filename: build_export_filename(config, nil, outer_batch_index, false), content: Base64.encode64(workbook.read_string) }
+        files << save_file(config, file)
+      end
+    end
+
+    files
+  end
+
+  # Gets data for this batch of patients that may not have already been present in the export config (such as specific symptoms).
+  def get_field_data(config)
+    data = config[:data].deep_dup
```

Comment:
Back to this again.
Can you test?
```
data = {}.merge!(config[:data])
data[:assessments] = config[:data][:assessments].deep_dup
```

This is faster than deep_dup over the whole hash and passes the test where the only-merge solution failed. Just want to see if it will continue to work through all test cases and further developments.

Reply:
This solution isn't working, as the assessments field data doesn't seem to be successfully deep_dup'd over